src/switchroot/ostree-prepare-root.c

```diff
@@ -44,6 +44,25 @@
 
 #include "ostree-mount-util.h"
 
+static void
+die (void)
+{
+  if (getpid() == 1)
+    {
+      /* We're init - Give the user a chance to recover */
+      fprintf (stderr, "init: Entering recovery shell\n");
```

Comment:
I'm wondering if this should be a build-time option rather than the default.  My main concern is systems where the owner has set it up for restricted/limited local access.  Think PoS terminal, computer lab etc.  Basically the same rationale applies to this that applies to the logic for some systems asking for a root password for fsck.

But we're not in a position to ask for root, so I think we need to make this part reliable :smile:   And stay locked by default, but I see the value in this for debugging.